app/dawn-hub/Views/OnboardingViews/Step3_ChooseDefaultSearchView/ChooseDefaultSearchView.jsx

```diff
@@ -0,0 +1,375 @@
+
+		this.fetchSearchEnginesAsync();
+	}
+
+	componentWillUnmount() {
+		document.removeEventListener('click', this.handleClickAway);
+	}
+
+	async fetchSearchEnginesAsync() {
+		// eslint-disable-next-line no-undef
+		if (typeof browser === 'undefined' || typeof browser.search === 'undefined') { // we are not in Dawn
+			this.setState(state => ({
+				...state,
+				otherSearchOptionsFetched: true,
+			}));
+			return;
+		}
+		// eslint-disable-next-line no-undef
+		const response = await browser.search.get(); // we are in Dawn, where this API is supported (unlike Chrome), and where the necessary search permission is added at build time (unlike non-Dawn builds of GBE)
+
+		// a successful response is guaranteed to be an array of search engine objects
+		// see https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/search/get
+		const otherOptions = response
+			.map(item => item.name)
+			// Two versions of the Startpage identifier are included for safety as it's unclear which version is / will be used
+			.filter(name => ![SEARCH_YAHOO, GLOW_BROWSER_SEARCH_GET_NAME, STARTPAGE_BROWSER_SEARCH_GET_NAME, SEARCH_STARTPAGE, SEARCH_BING].includes(name));
+
+		this.setState(state => ({
+			...state,
+			otherSearchOptionsFetched: true,
+			otherSearchOptions: otherOptions
+		}));
+	}
+
+	updateSelection = () => this.setState(prevState => (
+		{
+			chosenSearch: prevState.searchBeingConsidered,
+			searchBeingConsidered: null,
+			modalActive: false
+		}
+	));
+
+	cancelSelection = () => this.setState({ modalActive: false, searchBeingConsidered: null });
+
+	updateOtherSearchSelection = otherSelected => this.setState({ otherSearchSelected: otherSelected });
+
+	updateOtherListOpen = open => this.setState({ otherListOpen: open });
+
+	triggerConfirmationModal = selection => this.setState({ modalActive: true, searchBeingConsidered: selection });
+
+	handleClickAway = (e) => {
+		const { otherListOpen } = this.state;
+
+		const closeDropdownOnClickAway = (open, key, ref) => {
+			if (open && !ref.contains(e.target)) {
+				this.setState({ [`${key}`]: false });
+			}
+		};
+		closeDropdownOnClickAway(otherListOpen, 'otherListOpen', this.otherListRef);
+	}
+
+	handleSubmit = () => {
+		const { chosenSearch, otherSearchSelected } = this.state;
+		const { actions, history } = this.props;
+		const { setSetupStep, setDefaultSearch } = actions;
+
+		const chosenSearchName = chosenSearch === SEARCH_OTHER
+			? otherSearchSelected
+			: chosenSearch;
+
+		const payload = {
+			type: 'setDefaultSearch',
+			search: chosenSearchName,
+			isOther: chosenSearch === SEARCH_OTHER,
+		};
+
+		// The try/catch wrapper facilitates testing in non-Dawn browsers which have no search@ghostery.com extension
+		try {
+			chrome.runtime.sendMessage('search@ghostery.com', payload, () => {
+			});
+		} catch (error) {
+			console.log('Ilya sez: If you are seeing the following error in Dawn, please report it as a bug. Outside of Dawn, it is expected.');
```

Comment:
Use our logger instead of these console statements

Reply:
Done